tests/queries/test_query.py

```diff
@@ -23,6 +23,22 @@ def test_simple_query(self):
         self.assertEqual(lookup.rhs, 2)
         self.assertEqual(lookup.lhs.target, Author._meta.get_field('num'))
 
+    def test_simplecol_query(self):
+        def _test(child):
+            self.assertIsInstance(child.lhs, SimpleCol)
+            if isinstance(child.rhs, Expression):
+                self.assertIsInstance(child.rhs, SimpleCol)
+
+        query = Query(Author)
+        where = query.build_where(Q(num__gt=2, name__isnull=False) | Q(num__lt=F('id')))
+        for child in where.children:
+            with self.subTest(child=child):
+                if hasattr(child, 'children'):
+                    for grandchild in child.children:
+                        _test(grandchild)
+                else:
+                    _test(child)
+
```

Comment:
It feels like unrolling the loop by accessing `where.children[]` and inlining `_test` like in other tests do would make the actual assertions simpler to interpret in the case of future failures. I'll let the committer make the last call here though since the assertions are actually correct.

Reply:
not fully understood sorry. is this about the extracting `test_simplecol_query` content and applying to other tests as well?